crates/typos-cli/src/bin/typos-cli/main.rs

```diff
@@ -156,8 +159,24 @@ fn run_checks(args: &args::Args) -> proc_exit::ExitResult {
 
     let mut typos_found = false;
     let mut errors_found = false;
-    for path in args.path.iter() {
-        let cwd = if path == std::path::Path::new("-") {
+
+    let stdin_paths = if args.stdin_paths {
+        Some(
+            io::stdin()
+                .lines()
+                .map(|res| res.map(PathBuf::from))
+                .collect::<Result<_, _>>()
+                .with_code(proc_exit::sysexits::IO_ERR)?,
+        )
+    } else if args.stdin_paths0 {
+        Some(unix_read_paths_from_stdin()?)
+    } else {
+        None
+    };
+
+    for path in stdin_paths.as_ref().unwrap_or(&args.path) {
```

Comment:
I think I'd like a comment saying that the reason we can do `unwrap_or` is because these two sources are mutually exclusive